integrationtests/agent/adoption_test.go

```diff
@@ -317,71 +102,221 @@ var _ = Describe("Adoption", Label("adopt"), func() {
-			createConfigMap(
-				map[string]string{"foo": "bar"},
-				map[string]string{"foo": "234"},
-				map[string]string{"bar": "xzy"},
-			)
-			createBundleDeployment("adopt-random-metadata", true)
-			assertConfigMap(func(cm corev1.ConfigMap) error {
-				if err := isConfigMapAdopted(&cm); err != nil {
-					return err
-				}
-				return mapPartialMatch(cm.Data, map[string]string{"foo": "bar", "key": "value"})
+			env.createConfigMap(&corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{"bar": "xzy"},
+					Labels:      map[string]string{"foo": "234"},
+				},
+				Data: map[string]string{"foo": "bar"},
+			})
+			env.createBundleDeployment("adopt-random-metadata", true)
+			env.assertConfigMap(func(g Gomega, cm corev1.ConfigMap) {
+				env.assertConfigMapAdopted(g, &cm)
+				g.Expect(cm.Data).To(Equal(map[string]string{"foo": "bar", "key": "value"}))
+				g.Expect(cm.Annotations).To(HaveKeyWithValue("bar", "xzy"))
+				g.Expect(cm.Labels).To(HaveKeyWithValue("foo", "234"))
 			})
 		})
 	})
 
 	When("a bundle adopts a resource that is deployed by another bundle", Label("competing-bundles"), func() {
 		It("should complain about not owning the resource", func() {
-			createBundleDeployment("one", false)
-			waitForConfigMap("cm1")
-			createBundleDeployment("two", true)
-			assertBundleDeployment("one", bundleDeploymentNotOwnedByUs)
-			assertBundleDeployment("two", bundleDeploymentReady)
+			env.createBundleDeployment("one", false)
+			env.waitForConfigMap("cm1")
+			env.createBundleDeployment("two", true)
+			env.assertBundleDeployment("one", env.bundleDeploymentNotOwnedByUs)
+			env.assertBundleDeployment("two", env.bundleDeploymentReady)
 		})
 	})
 })
+
+type adoptEnv struct {
+	namespace string
+	env       *specEnv
+}
+
+func (e adoptEnv) createBundleDeployment(name string, takeOwnership bool) {
+	bundled := v1alpha1.BundleDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: clusterNS,
+		},
+		Spec: v1alpha1.BundleDeploymentSpec{
+			DeploymentID: "BundleDeploymentConfigMap",
+			Options: v1alpha1.BundleDeploymentOptions{
+				DefaultNamespace: e.namespace,
+				Helm: &v1alpha1.HelmOptions{
+					TakeOwnership: takeOwnership,
+				},
+			},
+		},
+	}
+
+	err := k8sClient.Create(context.TODO(), &bundled)
+	Expect(err).To(BeNil())
+	Expect(bundled).To(Not(BeNil()))
+	Expect(bundled.Spec.DeploymentID).ToNot(Equal(bundled.Status.AppliedDeploymentID))
+	Expect(bundled.Status.Ready).To(BeFalse())
+	Eventually(func(g Gomega) {
+		g.Expect(k8sClient.Get(
+			context.TODO(),
+			types.NamespacedName{Namespace: clusterNS, Name: name},
+			&bundled,
+		)).To(Succeed())
+		g.Expect(bundled.Status.Ready).To(BeTrue())
+	}).Should(Succeed(), "BundleDeployment not ready: status: %+v", bundled.Status)
+	Expect(bundled.Spec.DeploymentID).To(Equal(bundled.Status.AppliedDeploymentID))
+}
+
+func (e adoptEnv) waitForConfigMap(name string) {
+	time.Sleep(5 * time.Second)
```

Comment:
Why add a `Sleep` statement here? Shouldn't `Eventually` already cover any possible delay?

Reply:
Nice spotting. That shouldn't be there. I used it for debugging.